packages/dev/core/src/Lights/LTC/ltcTextureTool.ts

```diff
@@ -0,0 +1,109 @@
+import { RawTexture } from "core/Materials/Textures/rawTexture";
+import type { BaseTexture } from "core/Materials/Textures/baseTexture";
+import { Texture } from "core/Materials/Textures/texture";
+import type { Scene } from "core/scene";
+import { Constants } from "core/Engines/constants";
+import { Tools } from "core/Misc/tools";
+import type { Tuple } from "core/types";
+
+/**
+ * Interface that can provide LTC textures used by area Lights. Users can override the default ones to provide their own LTC textures.
+ */
+export interface IAreaLightLTCProvider {
+    /**
+     * Linearly trasnformed cossine texture for BRDF.
+     */
+    ltc1Texture: BaseTexture;
+
+    /**
+     * Linearly trasnformed cossine texture for BRDF.
+     */
+    ltc2Texture: BaseTexture;
+}
+
+/**
+ * Default provider for LTC textures. This provider will load the LTC data from the Babylon CDN.
+ */
+export class DefaultAreaLightLTCProvider implements IAreaLightLTCProvider {
+    /**
+     * LTC Texture 1 loaded from the Babylon.js CDN.
+     */
+    public ltc1Texture: BaseTexture;
+
+    /**
+     * LTC Texture 2 loaded from the Babylon.js CDN.
+     */
+    public ltc2Texture: BaseTexture;
+
+    constructor(scene: Scene) {
+        this._buildSceneLTCTextures(scene);
+    }
+
+    // Loads LTC textures from CDN and assigns them to ltc1Texture and ltc2Texture textures.
+    private async _buildSceneLTCTextures(scene: Scene): Promise<void> {
+        const textureData = await this._decodeLTCTextureData();
+        this.ltc1Texture = this._createLTCTextureFromArray(textureData[0], scene);
+        this.ltc2Texture = this._createLTCTextureFromArray(textureData[1], scene);
+        for (const mesh of scene.meshes) {
+            if (mesh.lightSources.filter((a) => a.getClassName() === "RectAreaLight")) {
+                mesh._markSubMeshesAsLightDirty();
+            }

```

Comment:
I think you are trying to see if there are light sources that are `RectAreaLight`. If so, this is slightly faster:
```suggestion
            if (mesh.lightSources.some((a) => a.getClassName() === "RectAreaLight")) {
                mesh._markSubMeshesAsLightDirty();
            }
```